src/Nancy/DefaultObjectSerializer.cs

```diff
@@ -17,18 +34,35 @@ public string Serialize(object sourceObject)
         {
             if (sourceObject == null)
             {
-                return String.Empty;
+                return string.Empty;
             }
 
-            var formatter = new BinaryFormatter();
+            dynamic serializedObject = (sourceObject is string)
+                ? sourceObject
+                : this.AddTypeInformation(sourceObject);
+
+            var json = SimpleJson.SerializeObject(serializedObject);
+            return Convert.ToBase64String(Encoding.UTF8.GetBytes(json));
+        }
 
-            using (var outputStream = new MemoryStream())
+        private dynamic AddTypeInformation(object sourceObject)
+        {
+            var sourceType = this.assemblyCatalog.GetAssemblies().Select(assembly => assembly.GetType(sourceObject.GetType().FullName)).FirstOrDefault(type => type != null);

```

Comment:
I think maybe AssemblyQualifiedName is empty if its a class in a separate poco lib or something.